### PR TITLE
tests: drivers: i2c: add generic sensor test with DT auto-detection

### PR DESCRIPTION
**Pull request by sudiiyengar-en**
Replace per-sensor test file with a single test_i2c_sensor.c that uses DT_HAS_COMPAT_STATUS_OKAY to compile tests only for sensors present in the devicetree. Core controller/target sources are excluded for sensor-only builds so the i2c_target alias is not required.

Each sensor overlay is self-contained with its own i2c-controller alias, avoiding conflicts with the core i2c.overlay. To add a new sensor, add a DT_HAS_COMPAT_STATUS_OKAY block in test_i2c_sensor.c, a DT overlay, and a testcase.yaml entry.

### CODE REVIEW

**Review by VeijoPesonen — DISMISSED** [review]

File: tests/drivers/i2c/src/test_i2c_sensor.c [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/kernel.h>
+#include <zephyr/ztest.h>
+#include <zephyr/dt-bindings/i2c/i2c.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/device.h>
+
+#include "test_i2c.h"
+
+
```
[the remark this answers is not in the document]
Comment (sudiiyengar-en) [reply tests/drivers/i2c/src/test_i2c_sensor.c]: Removing the Header test_i2c.h solves the need for an alias in the overlay

File: tests/drivers/i2c/src/test_i2c_sensor.c [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+ *
+ * Tests are compiled automatically for every sensor whose compatible
+ * string has status "okay" in the merged devicetree.  To add a new
+ * sensor, just:
+ *
+ *   1. Add a #if DT_HAS_COMPAT_STATUS_OKAY(vendor_part) block below
+ *      with the sensor-specific ZTEST test function.
+ *
+ *   2. Create boards/i2c_<sensor>.overlay with the sensor DT node
+ *      and an i2c-controller alias (see i2c_bme680.overlay).
```
[the remark this answers is not in the document]
Comment (sudiiyengar-en) [reply tests/drivers/i2c/src/test_i2c_sensor.c]: generic overlay only adding sensor test .

File: tests/drivers/i2c/README.rst [hunk 3]
```diff
@@ -40,7 +40,7 @@ Granular test suites:
 - ``CONFIG_I2C_BOUNDARY_TESTS``        Edge cases and limits
 - ``CONFIG_I2C_FAULT_INJECTION_TESTS`` Error handling validation
 - ``CONFIG_I2C_HARDWARE_AWARE_TESTS``  Platform-specific tests
-- ``CONFIG_I2C_BME680_TESTS``          BME680 sensor integration
+- ```CONFIG_I2C_SENSOR_TESTS``         Sensor integration (auto-detects DT sensors)
```
[the remark this answers is not in the document]
Comment (sudiiyengar-en) [reply tests/drivers/i2c/README.rst]: Addressed

File: tests/drivers/i2c/src/test_i2c_sensor.c [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+#include <zephyr/kernel.h>
+#include <zephyr/ztest.h>
+#include <zephyr/dt-bindings/i2c/i2c.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/device.h>
+
+#include "test_i2c.h"
+
+
+LOG_MODULE_REGISTER(alif_i2c_bme680, LOG_LEVEL_INF);
```
[the remark this answers is not in the document]
Comment (sudiiyengar-en) [reply tests/drivers/i2c/src/test_i2c_sensor.c]: Addressed

File: tests/drivers/i2c/README.rst [hunk 5]
```diff
@@ -63,37 +63,54 @@ Standard loopback (sync only, Standard frequency)
...
+Sensor integration (e.g. BME680)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: console
+
+   west build -p always -b alif_e8_dk/ae822fa0e5597xx0/rtss_hp \
+        tests/drivers/i2c -- \
+        -DDTC_OVERLAY_FILE=boards/i2c_sensor.overlay \
+        -DCONFIG_I2C_SENSOR_TESTS=y -DCONFIG_SENSOR=y
+
```
Comment (VeijoPesonen) [inline tests/drivers/i2c/README.rst]:
```suggestion
   west build -- -DDTC_OVERLAY_FILE=boards/i2c_sensor.overlay -DCONFIG_I2C_SENSOR_TESTS=y -DCONFIG_SENSOR=y

```

@sudiiyengar-en, @Anilkumar-Alifsemi Showing how to do a pristine build is documented on the Zephyr side already each and every sample's and or test's documentation does not need to repeat how to do it. Same goes for the target board selection as the supported targets are listed on the YAML files.

File: tests/drivers/i2c/README.rst [hunk 6]
```diff
@@ -63,37 +63,54 @@ Standard loopback (sync only, Standard frequency)
...
 
 .. code-block:: console
 
-   west build -p always -b alif_e7_dk tests/drivers/i2c \
+   west build -p always -b alif_e8_dk/ae822fa0e5597xx0/rtss_hp \
+        tests/drivers/i2c -- \
         -DDTC_OVERLAY_FILE=boards/i2c.overlay \
-        -- -DCONFIG_I2C_ALL_TESTS=y
+        -DCONFIG_I2C_ALL_TESTS=y
+
```
Comment (VeijoPesonen) [inline tests/drivers/i2c/README.rst]:
```suggestion
   west build -- \
       -DDTC_OVERLAY_FILE=boards/i2c.overlay \
        -DCONFIG_I2C_ALL_TESTS=y

```

**Review by VeijoPesonen — CHANGES_REQUESTED** [review]

File: tests/drivers/i2c/README.rst (hunk 5 above)
Comment (sudiiyengar-en) [reply tests/drivers/i2c/README.rst]: @VeijoPesonen  Readme should have full command that helps user to copy paste the build command. if i remove board and keep only arguments then keeping west build command is not useful. In appnote also we are giving for sample same this is for our test code . let me know if you still think its of repeated then i will change it .

Comment (VeijoPesonen) [reply tests/drivers/i2c/README.rst]: > Readme should have full command that helps user to copy paste the build command.

You need to modify the SKU value anyway if you have different variant of the chip - not everyone has the chip version ae822fa0e5597xx0.

> In appnote also we are giving for sample same this is for our test code.

When you build a sample we can make an assumption that the developer has read how to build one. They only should be informed what building this particular test case requires.

Comment (sudiiyengar-en) [reply tests/drivers/i2c/README.rst]: Addressed .

**Review by VeijoPesonen — DISMISSED** [review]

**Review by VeijoPesonen — APPROVED** [review]

**Review by Shamshoor-AlifSemi — APPROVED** [review]
